src/coreclr/jit/emitarm64.cpp

```diff
@@ -1012,32 +1012,32 @@ void emitter::emitInsSanityCheck(instrDesc* id)
                            // (predicated)
         case IF_SVE_CR_3A: // ........xx...... ...gggnnnnnddddd -- SVE extract element to SIMD&FP scalar register
             elemsize = id->idOpSize();
-            assert(insOptsScalableWithSimdScalar(id->idInsOpt())); // xx
```

Comment:
Hhm, so turns out, we won't be able to do the validation in `emitInsSanityCheck()`

Reply:
No, we've lost the information by this point.

What would be useful here would be if we had REG_Z1 which is distinct from REG_V1. This way we could assert `isSveVectorRegister()`.

AIUI, when predicate registers are added, REGNUM_BITS needs increasing from 6 bits to 7 bits. That would allow for 32 general + 32 neon + 16 SVE + 32 predicate.

There would then have to be a step somewhere around register allocations to convert SVE register numbers to/from NEON register numbers to ensure there is really only one set.

Alternatively, we could still have a distinct `REG_Z1`, but inside `emitIns_R_R_R()` it converts to `REG_V1` before storing to `_idReg1`. That means you can assert in `emitIns_R_R_R()` but not in `emitInsSanityCheck()`.


Either would allow for the removal of `INS_SCALABLE_OPTS_WITH_SIMD_SCALAR` too.